docs/root/start/sandboxes/route-mirror.rst

```diff
@@ -0,0 +1,146 @@
+.. _install_sandboxes_route_mirror:
+
+Route mirroring policies
+========================
+
+.. sidebar:: Requirements
+
+   .. include:: _include/docker-env-setup-link.rst
+
+This simple example demonstrates Envoy's request mirroring capability using
+`request mirror policies <https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-routeaction-requestmirrorpolicy>`__.
+
+Incoming requests are received by ``envoy-front-proxy`` service. The envoy instance running
+in this container is configured with two routes (:download:`front-envoy.yaml <_include/route-mirror/front-envoy.yaml>`):
+
+.. literalinclude:: _include/route-mirror/front-envoy.yaml
+   :language: yaml
+   :lines: 16-34
+   :linenos:
+   :emphasize-lines: 5-17
+   :caption: Envoy configuration with route mirror policy
+
+A request for the path ``/service/1`` is forwarded to the ``service1`` cluster.
+In addition, the request is also forwarded to the ``service1-mirror`` cluster.
+
+A request for the path ``/service/2`` is forwarded to the ``service2`` cluster.
+If a header ``x-mirror-cluster`` is specified in the request, envoy extracts the
+header value and forwards the request to a cluster with the same name, if found.
+For example, if we send a header with a request as ``x-mirror-cluster: service2-mirror``,
+the request will be forwarded to the ``service2-mirror`` cluster.
```

Comment:
```suggestion
Requests for the path ``/service/2`` are dynamically mirrored according to the presence and value of the ``x-mirror-cluster`` header.

All reqests for this path are  forwarded to the ``service2`` cluster, and are also mirrored to the cluster named in the header.

For example, if we send a header with a request as ``x-mirror-cluster: service2-mirror``,
the request will be forwarded to the ``service2-mirror`` cluster.
```

Im wondering if we need to say something about this pattern only being useful in a trusted environment - im not sure you would want an end-user determining where a request was mirrored, but i can see how it might be useful for a proxy to add this header for upstream processing, for example.

Reply:
Good point to mention. I will take a stab at it.